smsdk/client.py

```diff
@@ -225,3 +229,118 @@ def get_machine_schema(self, machine_source, types=[], return_mtype=False, **kwa
                             f"Unknow stat schema identified :: machine_type {machine_source} - "
                             f"title_prefix :: {stat.get('display', {}).get('title_prefix', '')}")
         return fields
+
+    def _get_factories(self, normalize=True, *args, **kwargs):
+        """
+        Get list of factories and associated metadata.  Note this includes extensive internal metadata.  
+
+        :param normalize: Flatten nested data structures
+        :type normalize: bool
+        :return: pandas dataframe
+        """
+        return self.get_data_v1('factory_v1', 'get_factories', normalize, *args, **kwargs)
+
+    def _get_machines(self, normalize=True, *args, **kwargs) -> pd.DataFrame:
+        """
+        Get list of machines and associated metadata.  Note this includes extensive internal metadata.  If you only want to get a list of machine names
+        then see also get_machine_names(). 
+
+        :param normalize: Flatten nested data structures
+        :type normalize: bool
+        :return: pandas dataframe
+        """
+        return self.get_data_v1('machine_v1', 'get_machines', normalize, *args, **kwargs)
+
+    def _get_machine_types(self, normalize=True, *args, **kwargs):
+        """
+        Get list of machine types and associated metadata.  Note this includes extensive internal metadata.  If you only want to get a list of machine type names
+        then see also get_machine_type_names(). 
+
+        :param normalize: Flatten nested data structures
+        :type normalize: bool
+        :return: pandas dataframe
+        """
+
+        return self.get_data_v1('machine_type_v1', 'get_machine_types', normalize, *args, **kwargs)
+    
+    def get_factories(self, normalize=True, *args, **kwargs):
+        generator = self._get_factories(normalize=normalize, *args, **kwargs)
+        data = []
+        for page in generator:
+            try:
+                data.append(page)
+            except Exception as e:
+                print(e)
+        data = pd.concat(data)
+        return data
+
+    def get_machines(self, normalize=True, *args, **kwargs):
+        generator = self._get_machines(normalize=normalize, *args, **kwargs)
+        data = []
+        for page in generator:
+            try:
+                data.append(page)
+            except Exception as e:
+                print(e)
+        data = pd.concat(data)
+        return data
+
+    def get_machine_types(self, normalize=True, *args, **kwargs):
+        generator = self._get_machine_types(normalize=normalize, *args, **kwargs)
+        data = []
+        for page in generator:
```

Comment:
there is repeating code, we should move this to single common class method.

Reply:
To maintain the symmetry with old code base